### PR TITLE
fix: resolve bad imports after build

### PR DESCRIPTION
**Pull request by nfantone**
- **fix**: solve `Cannot find module` errors after importing a function from the root.
- **chore**: generate typescript typings from jsdoc annotations into `types.d.ts`.

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -0,0 +1,691 @@
+/**
```
Comment (gaston-flores) [inline types.d.ts:1]: Maybe this should be added to `.gitignore`?

**Review by gaston-flores — APPROVED** [review]
Thank you! 🎉

File: types.d.ts (hunk 1 above)
Comment (nfantone) [reply types.d.ts:1]: Uhm... Could be 🤔. I think it's nice to have it as a quick go-to readme, though. We could maybe add a banner at the top of the file saying it was "auto-generated" and that it "should not be modified by hand"?

**Review by crash7 — APPROVED** [review]
Seems to solve the issue raised on https://github.com/flybondi/ramda-land/pull/20#issuecomment-762327573, thanks!

File: package.json [hunk 2]
```diff
@@ -40,19 +32,30 @@
...
     "docs": "jsdoc -c jsdoc.json",
-    "lint": "eslint index.js src/",
+    "lint": "eslint src/",
+    "prepack": "yarn build && yarn types",
     "prettier": "prettier --write src/ index.js",
     "test": "jest --env=node --colors --bail --detectOpenHandles --errorOnDeprecated --noStackTrace src/",
     "test:coverage": "yarn test --coverage --coverageDirectory .coverage",
-    "test:watch": "yarn test --watchAll"
+    "test:watch": "yarn test --watchAll",
+    "types": "jsdoc -t node_modules/tsd-jsdoc/dist -r src -d . && move-file types.d.ts index.d.ts && replace-in-files --regex='declare\\s+function' --replacement='export function' index.d.ts && replace-in-files --regex='^' --replacement='/*\\n  This file was auto-generated by tsd-jsdoc (https://github.com/englercj/tsd-jsdoc)\\n*/\\n\\n' index.d.ts"
```
Comment (nfantone) [inline package.json:58]: This is a bit scary, I know - what was the most succinct way I could put together to generate an `index.d.ts` file with a banner at the top and the appropriate `export` declarations, preserving portability (i.e.: no `sed`, no `awk`).

File: types.d.ts (hunk 1 above)
Comment (nfantone) [reply types.d.ts]: Added banner to `index.d.ts` (former `types.d.ts`).

**Pull request merged**